### ISSUE TITLE
Deeplinks are not working

### ISSUE THREAD

**Issue by herveweb** [Tlaster/PreCompose#228]
I've tried to setup deelinking using the `deeplink` field in the `scene` for my screen but it's not working for me. 

Any chance I can please get a link to some documentation on how to set it up?

**Comment by Tlaster** [comment]
Currently deeplink require manually handle on all platform, for example, in Android you can handle like this:
```Kotlin
override fun onCreate(savedInstanceState: Bundle?) {
    //....
    intent.data?.let {
        navigator.navigate(it.toString())
    }
    addOnNewIntentListener {
        it?.data?.let {
            navigator.navigate(it.toString())
        }
    }
}
```

**Comment by herveweb** [comment]
I believe it's kinda confusing when you see a `deepLinks` field in the `Scene` method. It makes us think that it's readily available for all platforms. But I guess this might be coming in the future.

<img width="452" alt="image" src="https://github.com/Tlaster/PreCompose/assets/11280628/1020cbde-3f03-4a05-a44f-4b27153f8587">

Unfortunately, for now I had to implement it separately on each platform. 
Thanks for the reply.

**Comment by Tlaster** [comment]
Yes I agree, the keyword makes people confused, I'm still trying to implement deeplink handling automatically.

**Comment by inveders** [comment]
> Currently deeplink require manually handle on all platform, for example, in Android you can handle like this:
> 
> ```kotlin
> override fun onCreate(savedInstanceState: Bundle?) {
>     //....
>     intent.data?.let {
>         navigator.navigate(it.toString())
>     }
>     addOnNewIntentListener {
>         it?.data?.let {
>             navigator.navigate(it.toString())
>         }
>     }
> }
> ```

Hello, An for ios how do you do that ?

**Comment by cbeyls** [comment]
> Currently deeplink require manually handle on all platform, for example, in Android you can handle like this:
> 
> ```kotlin
> override fun onCreate(savedInstanceState: Bundle?) {
>     //....
>     intent.data?.let {
>         navigator.navigate(it.toString())
>     }
>     addOnNewIntentListener {
>         it?.data?.let {
>             navigator.navigate(it.toString())
>         }
>     }
> }
> ```

People copy-pasting this code, please be careful not to send the deep link event multiple times on activity re-creation. Thus:

- You should read `intent.data` only once when `savedInstanceState` is null
- If you call `addOnNewIntentListener()` from a Composable function, make sure you only call it once and not on each recomposition. I would recommend using `DisposableEffect` to also unregister the listener when it leaves the composition.
